wasm/wasm.ir/src/org/jetbrains/kotlin/wasm/ir/Declarations.kt

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.wasm.ir
+
+
+class WasmModule(
+    val functionTypes: List<WasmFunctionType> = emptyList(),
+    val structs: List<WasmStructDeclaration> = emptyList(),
+
+    val importsInOrder: List<WasmNamedModuleField> = emptyList(),
+    val importedFunctions: List<WasmFunction.Imported> = emptyList(),
+    val importedMemories: List<WasmMemory> = emptyList(),
+    val importedTables: List<WasmTable> = emptyList(),
+    val importedGlobals: List<WasmGlobal> = emptyList(),
+
+    val definedFunctions: List<WasmFunction.Defined> = emptyList(),
+    val tables: List<WasmTable> = emptyList(),
+    val memories: List<WasmMemory> = emptyList(),
+    val globals: List<WasmGlobal> = emptyList(),
+    val exports: List<WasmExport<*>> = emptyList(),
+    val elements: List<WasmElement> = emptyList(),
+
+    val startFunction: WasmFunction? = null,
+
+    val data: List<WasmData> = emptyList(),
+    val dataCount: Boolean = false,
+)
+
+sealed class WasmNamedModuleField {
+    var id: Int? = null
+    open val name: String = ""
+}
+
+sealed class WasmFunction(
+    override val name: String,
+    val type: WasmFunctionType
+) : WasmNamedModuleField() {
+    class Defined(
+        name: String,
+        type: WasmFunctionType,
+        val locals: MutableList<WasmLocal> = mutableListOf(),
+        val instructions: MutableList<WasmInstr> = mutableListOf()
+    ) : WasmFunction(name, type)
+
+    class Imported(
+        name: String,
+        type: WasmFunctionType,
+        val importPair: WasmImportPair
+    ) : WasmFunction(name, type)
+}
+
+class WasmMemory(
+    val limits: WasmLimits = WasmLimits(1u, null),
```

Comment:
We actually don't use this default anymore. I'll remove it. It was a hack from times when compiler hardcoded memory limits instead of calculation the precise amount.

Reply:
Done